functions/components/metaTagInjector.ts

```diff
@@ -10,7 +10,11 @@ type MetaTagInjectorInput = {
  * to inject meta tags into the <head> of an HTML document.
  */
 export class MetaTagInjector implements HTMLRewriterElementContentHandlers {
-  constructor(private input: MetaTagInjectorInput) {}
+  blockedPathsList: string | null
+
+  constructor(private input: MetaTagInjectorInput, request: Request) {
+    this.blockedPathsList = request.headers.get('cf-blocked-paths-list')
+  }
```

Comment:
This can be simplified if you're willing to move `request.headers.get(...)` to the `append` body:

```suggestion
  constructor(private input: MetaTagInjectorInput, private request: Request) {}
```